docs/shadertrap_language.md

```diff
@@ -0,0 +1,338 @@
+
+### BIND_SHADER_STORAGE_BUFFER
+
+```
+BIND_STORAGE_BUFFER BUFFER buffer BINDING binding
+```
+
+Binds a buffer to a storage buffer binding point.
+- `buffer` is a buffer produced by `CREATE_BUFFER`
+- `binding` is a non-negative integer specifying a shader storage buffer binding point
+
+### BIND_TEXTURE
+
+```
+BIND_TEXTURE TEXTURE texture TEXTURE_UNIT unit
+```
+
+Binds a texture to a texture unit.
+- `texture` is a texture produced by `CREATE_EMPTY_TEXTURE_2D`
+- `unit` is a non-negative integer specifying which texture unit `texture` should be bound to
+
+To make a sampler uniform in a shader use the texture unit `unit` to which this texture has been bound, use `SET_UNIFORM` (supplying the value of `unit` as the value of the uniform).
+
+### BIND_UNIFORM_BUFFER
+
+```
+BIND_UNIFORM_BUFFER BUFFER buffer BINDING binding
+```
+
+Binds a buffer to a uniform buffer binding point.
+- `buffer` is a buffer produced by `CREATE_BUFFER`
+- `binding` is a non-negative integer specifying a uniform buffer binding point
+
+### COMPILE_SHADER
+
+```
+COMPILE_SHADER result SHADER shader
+```
+
+Compiles the shader associated with `shader`, which must be produced by `DECLARE_SHADER`. The compiled shader is identified by `result`.
+
+### CREATE_BUFFER
+
+```
+CREATE_BUFFER result SIZE_BYTES size INIT_VALUES
+  type_1 value_sequence_1
+  type_2 value_sequence_2
+  ...
+  type_n value_sequence_n
+```
+
+Creates a buffer of bytes that can be subsequently accessed by the GPU for various purposes, e.g. as a shader storage buffer, uniform buffer, vertex buffer or index buffer. The creation command does not specify for which purpose the buffer will later be used, and the buffer could in fact be used for multiple purposes, e.g. a compute shader could be used to modify the contents of the buffer, after which it could be fed to a graphics pipeline as vertex data.
+
+- The created buffer is identified by `result`
+- `size` is the size of the buffer in bytes
+- The `INIT_VALUES` parameter specifies the initial contents of the buffer as follows:
+  - Each of `type_1`, ..., `type_n` is one of `float`, `int`, `uint` or `byte`
+  - If `type_i` is `float` then `value_sequence_i` must be a sequence of floating-point literals.
+  - If `type_i` is `int` then `value_sequence_i` must be a sequence of 32-bit signed integer literals.
+  - If `type_i` is `uint` then `value_sequence_i` must be a sequence of 32-bit unsigned integer literals.
+  - If `type_i` is `byte` then `value_sequence_i` must be a sequence of byte literals (values in the range [0, 255]), and the length of `value_sequence_i` must be a multiple of 4
+  - The buffer is populated with the values provided by the value sequence in order. Each `float`, `int` and `uint` value occupies 4 bytes. Each `byte` value occupies 1 byte. There is no padding between elements.
```

Comment:
This line and the following could be removed from the list(`-`) as they are not dealing with the parameters / result but explaining points on the size calculation.